docs/contributor/testing-strategy.md

```diff
@@ -0,0 +1,103 @@
+- **Kubebuilder Test Framework and [EnvTest](https://book.kubebuilder.io/reference/envtest.html)**: For creating and executing integration tests that interact with the Kubernetes API.
+- **[Ginkgo](https://github.com/onsi/ginkgo) and [Gomega](https://github.com/onsi/gomega)**: For writing and executing unit tests with a BDD-style syntax and assertions.
+- **[k3d](https://k3d.io/)**: For creating short-living and lightweight Kubernetes clusters running within a Docker context.
+- **[Helm](https://helm.sh/)**: For deploying and managing test clusters and environments for end-to-end testing.
+- **[k6](https://k6.io/):**: For performance and stress testing.
+
+|Framework|Unit Testing|Integration Testing|End-to-End Testing|
+|--|--|--|--|
+|Golangci-lint| X | | |
+|Go-critic| X | | |
+|GoTest| X |  |  |
+|Kubebuilder Test Framework| X | X | |
+|EnvTest| X | X |  |
+|Ginkgo| X | X |  |
+|Gomega| X | X |  |
+|k3d|  |  | X |
+|Helm|  |  | X |
+|k6|  |  | X |
+
+
+## Test Automation
+
+The following CI/CD jobs are a part of the development cycle and executing quality assurance related steps:
+
+> **NOTE:** Jobs marked with `pull_request` are triggered with each pull request. Jobs marked with `push` are executed after the merge.
+
+- `golangci-lint / lint (pull_request/push)` - Is responsible for linting and static code analysis. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.golangci.yaml) and [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/golangci-lint.yaml).
+- `PR Markdown Link Check / markdown-link-check (pull_request)` - Checks if there are no broken links in the pull request `.md` files. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/mlc.config.json).
+- `Run unit tests / validate (pull_request/push)` - Executes basic create/update/delete functional tests of the reconciliation logic. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-tests.yaml).
+- `Run vuln check / test (pull_request/push)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-vuln-check.yaml).
+- `pull-infrastructure-manager-build` - Triggered with each pull request. It builds the Docker image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).	
+- `main-infrastructure-manager-build` - Triggered after the merge. Rebuilds the image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
```

Comment:
It is recommended to use more [descriptive text for links](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/06-links.md#links). How about: `For details, see the [configuration file](link)`?